app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php

```diff
@@ -408,17 +406,32 @@ public function getAllVoteablePresentations($summit_id)
         {
             $strategy = new RetrieveAllSummitVoteablePresentationsStrategy($this->repository, $this->event_repository, $this->resource_server_context);
             $response = $strategy->getEvents(['summit_id' => intval($summit_id)]);
+
+            $params  = [
+                'current_user' => $this->resource_server_context->getCurrentUser(true),
+                'use_cache' => true,
+            ];
+
+            $filter = $strategy->getFilter();
+
+            if(!is_null($filter)) {
+                $votingDateFilter = $filter->getFilter('presentation_attendee_vote_date');
+                if ($votingDateFilter != null) {
+                    $params['begin_attendee_voting_period_date'] = $votingDateFilter[0]->getValue();
+                    if (count($votingDateFilter) > 1) {
+                        $params['end_attendee_voting_period_date'] = $votingDateFilter[1]->getValue();
+                    }
+                }
+            }
+
             return $this->ok
             (
                 $response->toArray
                 (
                     self::getExpands(),
                     self::getFields(),
                     self::getRelations(),
-                    [
-                        'current_user' => $this->resource_server_context->getCurrentUser(true),
-                        'use_cache' => true,
-                    ],
+                    $params,
                     $this->getSerializerType()
```

Comment:
we need to defined the logic to get the proper serializer here on getSerializerType based on user rites

Reply:
new serializer type getter was created to cover this scenario